engine/src/main/java/org/terasology/utilities/procedural/SimplexNoise.java

```diff
@@ -154,8 +169,8 @@ public float noise(float xin, float yin) {
         float y2 = y0 - 1.0f + 2.0f * G2;
 
         // Work out the hashed gradient indices of the three simplex corners
-        int ii = i & 255;
-        int jj = j & 255;
+        int ii = i % permCount;
```

Comment:
In the Perlin implementation you did some combination of `floorMod` and `floor`  - why is a simple modulo operation sufficient here, and why is it not sufficient in Perlin implementation?

Reply:
oh I thought that here `i`, `j` are positive integers but on second look I can see that this is problematic, fixed this